src/Adapter/Module/AdminModuleDataProvider.php

```diff
@@ -172,8 +172,9 @@ public function isAllowedAccess($action, $name = '')
         return ($this->employee->can('edit', 'AdminModulessf') && $this->moduleProvider->can('configure', $name));
     }
 
-    public function generateAddonsUrls(array $addons, $specific_action = null)
+    public function generateAddonsUrls(AddonsCollection $addons, $specific_action = null)
     {
+        $addons = $addons->toArray();
```

Comment:
Is this line needed when you implement `Iterator` in the concerned class? :)

Reply:
i prefer being explicit on what I retrieve (a simple array)